registry/service.go

```diff
@@ -13,29 +13,41 @@ import (
 	registrytypes "github.com/docker/engine-api/types/registry"
 )
 
-// Service is a registry service. It tracks configuration data such as a list
+// Service is the interface defining what a registry service should implement.
+type Service interface {
+	Auth(authConfig *types.AuthConfig, userAgent string) (status, token string, err error)
```

Comment:
@stevvooe hum why not, but I wonder how I should do it (i.e. just defining in the interface/methods and not using it inside letting other ppl using it or going through the implementation too). I'll take a look :wink: